File: services/dd-ctf/backend/app/schemas/category.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class CategoryBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    description: Optional[str] = None


class CategoryCreate(CategoryBase):
    pass


class CategoryRead(CategoryBase):
    id: int
    created_at: datetime


class CategoryUpdate(CategoryBase):
    id: int

File: frontend/types/category.ts
export interface CategoryBase {
    name: string;
    description?: string;
}

// export interface CategoryCreate extends CategoryBase {
//     _type?: 'create';
// }

export interface CategoryRead extends CategoryBase {
    id: number;
    created_at: Date;
}

export interface CategoryUpdate extends CategoryBase {
    id: number;
}
